src/screens/LoginScreen/index.js

```diff
@@ -3,24 +3,29 @@ import { useDispatch } from 'react-redux';
 import { Text, View, Button } from 'react-native';
 import { object } from 'prop-types';
 
-import LoginForm from 'components/user/LoginForm';
+import LoginForm from 'components/LoginForm';
 import { login } from 'actions/userActions';
-import useNavigateOnLoginEffect from 'hooks/useNavigateOnLoginEffect';
 import translate from 'utils/i18n';
-import { SIGN_UP_SCREEN } from 'constants/screens';
+import { SIGN_UP_SCREEN, MAIN_SCREEN } from 'constants/screens';
 import styles from './styles';
 
 const LoginScreen = memo(({ navigation }) => {
   const dispatch = useDispatch();
-  const loginRequest = useCallback(user => dispatch(login(user)), [dispatch]);
+  const loginRequest = useCallback(
+    async user => {
+      await dispatch(login(user));
+      navigation.navigate(MAIN_SCREEN);
+    },
+    [dispatch],
+  );
 
-  useNavigateOnLoginEffect(navigation);
```

Comment:
Why did you remove the use of this hook?

Reply:
I was getting some really weird loops when logging in due to how the section actions are dispatched.